redfish-core/lib/assembly.hpp

```diff
@@ -252,29 +252,37 @@ void getAssemblyPresence(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
                          const auto& serviceName, const auto& assembly,
                          const auto& assemblyIndex)
 {
-    nlohmann::json& assemblyArray = asyncResp->res.jsonValue["Assemblies"];
-    nlohmann::json& assemblyData = assemblyArray.at(assemblyIndex);
-
-    assemblyData["Status"]["State"] = "Enabled";
-
     sdbusplus::asio::getProperty<bool>(
         *crow::connections::systemBus, serviceName, assembly,
         "xyz.openbmc_project.Inventory.Item", "Present",
-        [asyncResp, assemblyIndex](const boost::system::error_code& ec,
-                                   const bool value) {
+        [asyncResp, assemblyIndex,
+         assembly](const boost::system::error_code& ec, const bool value) {
         if (ec)
         {
             BMCWEB_LOG_ERROR("DBUS response error: {}", ec.value());
             messages::internalError(asyncResp->res);
             return;
         }
 
+        nlohmann::json& array = asyncResp->res.jsonValue["Assemblies"];
+        nlohmann::json& data = array.at(assemblyIndex);
+        data["Status"]["State"] = "Enabled";
```

Comment:
Why is it moving to inside `getProperty()`?

Isn't this "Enabled" a default state ?
If so, should it be set before getProperty()?

Reply:
we can do 2 things,
Either move these lines inside get property.
OR else , keep it on top, and pass those variable to lambda, and make use of it.
Currently I see we having duplicate storage of data and array

Reply:
We need a default value for  -> assemblyData["Status"]["State"] = "Enabled";
in case getProperty fails. 
@Alpana07 The change you have done fails the purpose here.
Instead, your current implementation can move under the (!value) check and at top can have a default value for ReadyToRemove = false.